docs/src/writing_good_rules.md

```diff
@@ -102,6 +102,18 @@ function rrule(::typeof(*), A::AbstractMatrix, B::AbstractMatrix)
 end
 ```
 
+!!! note "It is often good to `@thunk` your projections"
+    The above example is potentially a good place for using a [`@thunk`](@ref).
+    This is not required, but can in some cases be more computationally efficient, see [Use `Thunk`s appropriately](@ref).
+    When combining thunks and projections, `@thunk()` must be the outermost call.
+
```

Comment:
@oxinabox I left out
```
`project` itself will when applied to `AbstractThunk`s make a deferred operation, rather than unthunking directly.
    But this is not the case for most operations such as `*`
```
which didn't entirely make sense to me, and also didn't seem super important (it's superseded by always having the `@thunk` on the outside anyways). If you think this _should_ get included, happy to figure out with you how to make it clear to others:)

Reply:
i am happy to exclude it